game/SwitchDB_Miidatafile.ksy

```diff
@@ -0,0 +1,157 @@
+  - id: mouth_stretch
+    type: b3
+    doc: Mouth stretch. Ranges from 0 to 6, small to big.
+  - id: nose_vertical
+    type: b5
+    doc: Nose Y (vertical) position. Ranges from 18 to 0, low to high.
+  - id: eyebrow_stretch
+    type: b3
+    doc: Eyebrow stretch. Ranges from 0 to 6, small to big.
+  - id: mouth_vertical
+    type: b5
+    doc: Mouth Y (vertical) position. Ranges from 18 to 0, low to high.
+  - id: eye_rotation
+    type: b3
+    doc: Eye rotation. Ranges from 0 to 7, down to up. Note that some eye types have a default rotation.
+  - id: facial_hair_vertical
+    type: b5
+    doc: Mustache Y (vertical) position. Ranges from 22 to 0, low to high.
+  - id: eye_stretch
+    type: b3
+    doc: Eye stretch. Ranges from 0 to 6, small to big.
+  - id: glasses_vertical
+    type: b5
+    doc: Glasses Y (vertical) position. Ranges from 20 to 0, low to high.
+  - id: eye_size
+    type: b3
+    doc: Eye size. Ranges from 0 to 7, small to big.
+  - id: mole_horizontal
+    type: b5
+    doc: Mole X (horizontal) position. Ranges from 0 to 16, left to right.
+  - id: mole_vertical
+    type: u1
+    doc: Mole Y (vertical) position. Ranges from 30 to 0, low to high.
+  - id: glasses_type
+    type: u1
+    doc: Glasses type. Ranges from 0 to 19. Not ordered the same as visible in editor.
+  - id: face_type
+    type: b4
+    doc: Face shape. Ranges from 0 to 11. Not ordered the same as visible in editor.
+  - id: favorite_color
+    type: b4
+    doc: Favorite color. Ranges from 0 to 11.
+  - id: face_wrinkles
+    type: b4
+    doc: Face wrinkles. Ranges from 0 to 11.
+  - id: face_color
+    type: b4
+    doc: Skin color. Ranges from 0 to 9. Not ordered the same as visible in editor.
+  - id: eye_horizontal
+    type: b4
+    doc: Eye X (horizontal) distance. Ranges from 0 to 12, close to far.
+  - id: face_makeup
+    type: b4
+    doc: Face makeup. Ranges from 0 to 11.
+  - id: eyebrow_rotation
+    type: b4
+    doc: Eyebrow rotation. Ranges from 0 to 11, down to up. Note that some eye types have a default rotation.
+  - id: eyebrow_size
+    type: b4
+    doc: Eyebrow size. Ranges from 0 to 8, small to big.
+  - id: eyebrow_vertical
+    type: b4
+    doc: Eyebrow Y (vertical) position. Ranges from 15 to 0, low to high.
+  - id: eyebrow_horizontal
+    type: b4
+    doc: Eyebrow X (horizontal) distance. Ranges from 0 to 12, close to far.
+  - id: mouth_size
+    type: b4
+    doc: Mouth size. Ranges from 0 to 8, small to big.
+  - id: nose_size
+    type: b4
+    doc: Nose size. Ranges from 0 to 8, small to big.
+  - id: mole_size
+    type: b4
+    doc: Mole size. Ranges from 0 to 8, small to big.
+  - id: facial_hair_size
+    type: b4
+    doc: Mustache size. Ranges from 0 to 8, small to big.
+  - id: mii_name
+    type: str
+    size: 20
+    encoding: utf-16le
+    doc: Mii name. Can be up to 10 characters long.
+  - id: unknown
+    type: u1
+    repeat: expr
+    repeat-expr: 16

```

Comment:
Instead of an array of `u1` values, it's almost always better to use `size` instead, to create a simple byte array.
```suggestion
    size: 16
```